src/main/java/utilities/Parser.java

```diff
@@ -0,0 +1,103 @@
+package utilities;
+import java.util.Scanner;
+
+import command.Command;
+import command.ExitCommand;
+import command.ListCommand;
+import command.MarkCommand;
+import command.UnmarkCommand;
+import command.DeleteCommand;
+import command.TaskCommand;
+import command.FindCommand;
+import exception.FormatException;
+import exception.NoInputException;
+import task.TaskList;
+
+/**
+ * Parser class is used to parse user input and execute commands.
+ */
+public class Parser {
+    /**
+     * Static method to add horizontal lines and indentation to the dialog.
+     * @param dialog The dialog to be formatted.
+     * @return The formatted dialog.
+     */
+    public static String addHorizontalLinesAndIndentation(String dialog) {
```

Comment:
I'm not sure if this function name is considered too long under "code quality" rules. Something shorter could be a name like "formatDialog()" but that would miss out on explaining what the function does explicitly.